lib/upi-consumer.coffee

```diff
@@ -263,7 +279,7 @@ class UPIConsumer
         else
           throw new Error('Got neither type nor info')
       text:
-        text: "#{if type?.text?.text then ':: '+type.text.text+'\n' else ''}#{info?.text?.text ? ''}"
+        text: (Object.keys(doc).join(", ") + "\nBLERG 0: " + doc[Object.keys(doc)[0]] + "\nBLERG 1: " + doc[Object.keys(doc)[1]] + "\nBLERG 1 keys: " + Object.keys(doc[Object.keys(doc)[1]]) + "\nBLERG 1[0]: " + doc[Object.keys(doc)[1]]["text"] + "--(Karl was here)\n") + "#{if type?.text?.text then ':: '+type.text.text+'\n' else ''}#{info?.text?.text ? ''}"
```

Comment:
@kostmo, thank you for your work, but *please* file new issues against relevant packages or at least ide-haskell in the future. It's hard enough to maintain this mess as it is.

So second point is an oversight on my part, should be fixed by ide-haskell-hasktags-0.0.6.

Third point seems to be a problem with ghc-mod -- it seems it doesn't resolve import aliases to files. Can't tell from the top of my head if it's easy to fix, or even fixable in principle -- it's very probable that it's a limitation of GHC API.